src/history/cam_hist_file.F90

```diff
@@ -1556,13 +1570,25 @@ subroutine config_write_field(this, field, split_file_index, restart, &
          call pio_setframe(this%hist_files(split_file_index), varid, int(sample_index,kind=PIO_OFFSET_KIND))
          buff_ptr => field%buffers
          if (frank == 1) then
-            call hist_buffer_norm_value(buff_ptr, field_data(:,1))
-            call cam_grid_write_dist_array(this%hist_files(split_file_index), field_decomp, dim_sizes(1: frank), &
-                 field_shape, field_data(:,1), varid)
+            if (trim(field_precision) == 'REAL32') then
+               call hist_buffer_norm_value(buff_ptr, field_data_r4(:,1))
+               call cam_grid_write_dist_array(this%hist_files(split_file_index), field_decomp, dim_sizes(1: frank), &
+                    field_shape, field_data_r4(:,1), varid)
+            else
+               call hist_buffer_norm_value(buff_ptr, field_data_r8(:,1))
+               call cam_grid_write_dist_array(this%hist_files(split_file_index), field_decomp, dim_sizes(1: frank), &
+                    field_shape, field_data_r8(:,1), varid)
+            end if
          else
-            call hist_buffer_norm_value(buff_ptr, field_data)
-            call cam_grid_write_dist_array(this%hist_files(split_file_index), field_decomp, dim_sizes(1: frank), &
-                 field_shape, field_data, varid)
+            if (trim(field_precision) == 'REAL32') then
+               call hist_buffer_norm_value(buff_ptr, field_data_r4)
+               call cam_grid_write_dist_array(this%hist_files(split_file_index), field_decomp, dim_sizes(1: frank), &
+                    field_shape, field_data_r4, varid)
+            else
+               call hist_buffer_norm_value(buff_ptr, field_data_r8)
+               call cam_grid_write_dist_array(this%hist_files(split_file_index), field_decomp, dim_sizes(1: frank), &
```

Comment:
Very silly nit-pick but can we remove the space between `1:` and `frank` here?

```suggestion
               call cam_grid_write_dist_array(this%hist_files(split_file_index), field_decomp, dim_sizes(1:frank), &
                    field_shape, field_data_r4, varid)
            else
               call hist_buffer_norm_value(buff_ptr, field_data_r8)
               call cam_grid_write_dist_array(this%hist_files(split_file_index), field_decomp, dim_sizes(1:frank), &
```

Reply:
the picked nit has been squashed (?)